lib/truffle/truffle/patching.rb

```diff
@@ -1,140 +1,69 @@
-require 'pathname'
-
 module Truffle::Patching
-  DIR     = Pathname(Truffle::Boot.ruby_home).join("lib/patches")
-  # Allowed operations are: :before, :after, :instead, :ignore
-  PATCHES = { "bundler"                                 => :before,
-              "bundler/version"                         => :after,
-              "bundler/cli/exec"                        => :after,
-              "bundler/compact_index_client/updater"    => :after,
-              "bundler/current_ruby"                    => :after,
-              "bundler/dependency"                      => :after,
-              "bundler/fetcher/downloader"              => :after,
-              "bundler/fetcher/compact_index"           => :after,
-              "bundler/ruby_version"                    => :after,
-              "bundler/source/rubygems"                 => :after,
-
-              "rubygems"                                => :before,
-              "rubygems/ext"                            => :after,
-              "rubygems/package"                        => :after,
-              "rubygems/remote_fetcher"                 => :after,
-              "rubygems/request"                        => :after,
-              "rubygems/request_set/gem_dependency_api" => :after,
-
-              "rspec/support"                           => :after,
-
-              "unf/normalizer_cruby"                    => :before,
-  }
-
-  PATCHES.each do |file, _|
-    unless DIR.join(file).sub_ext('.rb').exist?
-      raise "file #{file} is missing in #{DIR}"
+  extend self
+
+  def patches
+    @patches ||= begin
+      require 'pathname'
+      Pathname.glob(dir.join('*')).reduce({}) do |patches, path|
+        next patches unless path.directory?
+        patches.update path.basename.to_s => path
+      end
```

Comment:
I had considered your proposal (the second snippet) and I don't find it better than:
```ruby
require 'pathname'
Pathname.glob(dir.join('*')).each_with_object({}) do |path, patches|
   patches[path.basename.to_s] = path if path.directory?
end
```
All those things you have listed I've also already considered, this particular code is called once for less then 10 directories (which may grow to few tens tops), therefore these little optimisations have hardly an observable impact. This particular file only uses `pathname` API, single API is simpler than more, I don't want to mix with second `File.*` API.

So far I did not recognise any technical reason to change it, in that case I would have already done it. The remaining are different personal subjective preferences which I don't think are enough to change the code. Therefore I've considered your proposal, but unless there is a technical reason for change, Thanks but sorry I am not accepting your proposal.